R/load_data.R

```diff
@@ -0,0 +1,96 @@
+# Helper functions not for export
+
+#' Check for a version of data to load
+#'
+#' This function determines if a user is requesting the latest version
+#' of the bundled data or wants to access one of the earlier veresions.
+#'
+#' @param mode Determines which data to handle. Defaults to somatic_hypermutation_locations. Will grow with more options as more data is version tracked.
+#' @param this_genome_build The genome build of the data if coordinate-based. Accepts grch37 (default) or hg38.
+#'
+#' @return data frame
+#' @import config dplyr readr GAMBLR.data
+#'
+#' @examples
+#' determine_version()
+#' determine_version(this_genome_build = "hg38")
+#'
+determine_version <- function(
+    mode = "somatic_hypermutation_locations",
+    this_genome_build = "grch37"
+){
+    # Determine the latest version of the data
+    # Get absolute paths to bundled files in GAMBLR.data
+    all_files <- system.file(
+        "extdata",
+        package = "GAMBLR.data"
+    ) %>%
+    list.files(
+        recursive = TRUE,
+        full.names = TRUE
+    )
+
+    # Extract version from the path
+    all_files <- gsub(
+        ".*extdata/",
+        "",
+        all_files
+    )
+    all_files <- all_files[grepl(mode, all_files)]
+
+    # Determine the highst version
+    versions <- gsub(".*[/]([^/]+)[/].*", "\\1", all_files)
+
+    versions <- sort(
+        numeric_version(
+            versions
+            )
+    )
+
+    latest_version <- max(versions)
+
+    # Which version did the user requested in config?
+    requested_version <- config::get("bundled_data_versions")[[mode]]
```

Comment:
Right! I'll fix this!

Reply:
When I started adding other objects and looking at this, I realized here that we would have this in plural - because different data types develop more than the others and iterate through versions differently. For example, for aSHM sites we already have versions 0, 1, 2, and 3 - but for lymphoma genes only 0 and 1. So config here will have different keys depending on the data type.